src/views/activity/activity_item_view.js

```diff
@@ -20,5 +26,26 @@ export default BaseView.extend({
 
   sendSelectionDetails (event) {
     webChannel.sendUrlSelected(this.model.url, 'url');
+  },
+
+  injectFavicon () {
+    // Manually construct the favicon element and inject it favicon into the
+    // DOM, allowing us to delegate events to its onLoad event, which is not
+    // possible through Ampersand since load events do not bubble.
+    const favicon = document.createElement('img');
+    favicon.src = this.model.displayImage();
+    favicon.onload = this.calculateIconColor;
```

Comment:
You're getting warmer ;-)
- Need to set handlers before you set the `src`, otherwise you will be racing the load/error events
- Missing a `favicon.onerror` handler
- Have a peek at the first SO answer here http://stackoverflow.com/questions/9815762/detect-when-an-image-fails-to-load-in-javascript